In [2]:
import pandas as pd
import numpy as np

In [6]:
bb_df = pd.read_csv('baseball_data.csv')

In [8]:
bb_df.head(10)

name handedness  height  weight    avg   HR
0       Tom Brown          R      73     170  0.000    0
1  Denny Lemaster          R      73     182  0.130    4
2       Joe Nolan          L      71     175  0.263   27
3     Denny Doyle          L      69     175  0.250   16
4   Jose Cardenal          R      70     150  0.275  138
5       Mike Ryan          R      74     205  0.193   28
6  Fritz Peterson          B      72     185  0.159    2
7    Dick Bertell          R      72     200  0.250   10
8      Rod Kanehl          R      73     180  0.241    6
9    Ozzie Osborn          R      74     195  0.000    0

From this we can see that we have five columns of data and a sixth for an identifier. Handedness looks to be whether a person was left-handed, right-handed or ambidextrous. Height looks to be in inches. Weight looks to be in pounds. AVG, from the data dictionary, is the batting average for that person. HR is the amount of home runs for that person.

In [9]:
bb_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1157 entries, 0 to 1156
Data columns (total 6 columns):
name          1157 non-null object
handedness    1157 non-null object
height        1157 non-null int64
weight        1157 non-null int64
avg           1157 non-null float64
HR            1157 non-null int64
dtypes: float64(1), int64(3), object(2)
memory usage: 54.3+ KB


We can see that our dataset contains 1157 values. 

In [12]:
bb_df.groupby('handedness')['name'].count()

handedness
B    104
L    316
R    737
Name: name, dtype: int64

From this, we can see that around 104 (8.99%) people are ambidextrous, 316 (27.31%) are left handed while the rest, 737 (63.70%), are right handed. 

In [22]:
bb_df.groupby('height').mean()

weight       avg          HR
height                                  
65      148.000000  0.242000   41.000000
66      162.500000  0.254500   18.000000
67      159.250000  0.263250  100.000000
68      165.545455  0.229818   14.181818
69      165.463415  0.236122   26.463415
70      170.758929  0.230643   39.491071
71      175.965517  0.208186   47.213793
72      181.271552  0.212819   42.301724
73      184.360825  0.192660   50.974227
74      190.811111  0.179272   61.138889
75      195.330579  0.147298   49.570248
76      201.106061  0.104288   34.000000
77      204.100000  0.047700    1.700000
78      216.272727  0.046182   40.181818
79      216.000000  0.085333   13.666667
80      225.500000  0.042000    2.500000

Whoa. Why does it look like those who are 67 inches (5' 7") hit the most home runs? Are they just physically more capable and therefore apt at baseball? Debatable. Let's dive deeper.

In [25]:
bb_df[bb_df['height']==67]

name handedness  height  weight    avg   HR
471   Vic Davalillo          L      67     150  0.279   36
638      Don Buford          B      67     160  0.264   93
672      Joe Morgan          L      67     160  0.271  268
1065      Ron Brand          R      67     167  0.239    3

It looks like one man, Joe Morgan, actually hit the most of these home runes and is carrying the other 67" men. If we remove him from the (for the sake of comparisons)

In [35]:
bb_df[bb_df['HR']<500]['HR'].max()

475